### DISCUSSION TITLE
Update signature in server JSON

### DISCUSSION THREAD

**Discussion by jb-alvarado** [tauri-apps/tauri#6106] [Q&A] [answered]
Hello,
I very new to tauri and found now the updater function. I have already setup my github action, to create releases and upload them. But now I would like to automate the passing of the right signature. 

Is it possible to create the JSON like:

```JSON
{
    "version": "v0.1.4",
    "notes": "",
    "pub_date": "2023-01-19T10:48:21Z",
    "platforms": {
        "darwin-x86_64": {
            "signature": "https://github.com/user/repo/releases/download/v0.1.4/my_app.app.tar.gz.sig",
            "url": "https://github.com/user/repo/releases/download/v0.1.4/my_app.app.tar.gz"
        },
        "linux-x86_64": {
            "signature": "https://github.com/user/repo/releases/download/v0.1.4/my_app_v0.1.4_amd64.AppImage.tar.gz.sig",
            "url": "https://github.com/user/repo/releases/download/v0.1.4/my_app_v0.1.4_amd64.AppImage.tar.gz "
        },
        "windows-x86_64": {
            "signature": "https://github.com/user/repo/releases/download/v0.1.4/my_app_0.1.4_x64_en-US.msi.zip.sig",
            "url": "https://github.com/user/repo/releases/download/v0.1.4/my_app_0.1.4_x64_en-US.msi.zip"
        }
    }
}
```

Or how do you fill this signature automatically?

**Comment by FabianLars** [comment, marked as answer]
If you use tauri-action from git (via `tauri-apps/tauri-action@dev`) it can automatically generate this file for you, there mayyy be some bugs that still need to be ironed out.

Also, the signature is **not** an url or path to the .sig, but the actual content of that file as a string!

**Reply by jb-alvarado** [reply]
Thank you for your replay! Yes I know, that this *.sig files are generated automatically, they are also uploaded to my release page.

But at the moment, I have to download this *.sig files, open them in a text editor and copy the string to the JSON. This step would be nice, if there is a way to automate it :-).

**Reply by FabianLars** [reply]
> This step would be nice, if there is a way to automate it :-).

Yeah like i said, you can use tauri-action's dev branch which does that for you, or you can read it's source code to write your own script that can do this. https://github.com/tauri-apps/tauri-action/blob/dev/packages/action/src/upload-version-json.ts

**Reply by jb-alvarado** [reply]
Ah, sorry - then I got it wrong. Thank you!
